### DISCUSSION TITLE
Persistent "Setting Metals Java Home will result in updating Bloop's global Json file..." popup

### DISCUSSION THREAD

**Discussion by dzanot** [scalameta/metals-vscode#1038] [Q&A] [answered]
Howdy,

I am experiencing an issue after importing a project into metals. It seemingly works (I can jump to definition for example) but when I try to enable the `Toggle showing implicit parameters` setting, I get this pop up:
![Screen Shot 2022-06-24 at 1 07 02 PM](https://user-images.githubusercontent.com/5342283/175660040-d6dc2172-d267-4208-9df4-d660453adb00.png)
Pressing `Apply and restart bloop` dismisses the modal, but the next time I try to show implicit hints the modal reappears

I have tried `killall java` along with removing all the metals/bsp/bloop directories, and re-importing, but the issue persists.

I am using metals with versions:
Bloop: v1.5.0-18-003e6c7b
Java: 11.0.9 
Server: 0.11.6

Related (I think): https://github.com/scalameta/nvim-metals/discussions/411 https://github.com/scalameta/metals/issues/4001

Please let me know if an Issue is more appropriate and I will create one. Thanks

**Comment by dzanot** [comment, marked as answer]
PS: tried setting server version to `latest.snapshot` per the suggestion in the nvim discussion, but that started a similar loop but for setting/installing the java-home location

**Reply by tgodzik** [reply]
This should have actually been fixed. Could you try setting the Metals version to a prerelease? We plan to do a new full release soon so it would be good to make sure this is fixed.

**Reply by dzanot** [reply]
Set the metals server version to the latest I saw (`0.11.6+143-5bd656f1-SNAPSHOT`) and enabled pre-release on the VSCode metals plugin, and that fixed the issue. Thanks @tgodzik !

**Reply by kpodsiad** [reply]
@dzanot take into account that pre-release might result in less stable experience so you mant want to switch a stable version of VSCode metals extension after next release ;)

**Reply by dzanot** [reply]
Good to know @kpodsiad, I'll keep that in mind, and keep an eye on the release. Thanks :D
